sklearn/mixture/_gaussian_mixture.py

```diff
@@ -845,8 +851,9 @@ def _set_parameters(self, params):
         # Attributes computation
         _, n_features = self.means_.shape
 
+        dtype = self.precisions_cholesky_.dtype
         if self.covariance_type == "full":
-            self.precisions_ = np.empty(self.precisions_cholesky_.shape)
+            self.precisions_ = np.empty(self.precisions_cholesky_.shape, dtype=dtype)
```

Comment:
I think we could use `empty_like(self.precisions_cholesky_)` here. Just as a convenience really